### PR TITLE
HUB-942: Upgrade the hub to use Dropwizard 2

### PR DESCRIPTION
**Pull request by Wynndow**
I've been tinkering with this in the background for a while. All tests pass, including the acceptance tests. 
The **VAST** majority of this PR is boilerplate changes in the tests.
It's **strongly** recommended to review this commit by commit. There are extra details in the commit messages.

### Dependent PRs, in order of requirement
https://github.com/alphagov/dropwizard-logstash/pull/36
https://github.com/alphagov/verify-test-utils/pull/19
https://github.com/alphagov/verify-utils-libs/pull/63
https://github.com/alphagov/verify-saml-libs/pull/131
https://github.com/alphagov/verify-event-emitter/pull/49

### Bumps dropwizard to version 2
The versions of the dropwizard dependencies are taken to 2.0.21, and any issues as described in the [upgrade notes](https://www.dropwizard.io/en/latest/manual/upgrade-notes/upgrade-notes-2_0_x.html) are addressed.

### Bumps OpenSAML to 3.4.6
The old version didn't work with the new version of dropwizard. There was a clash of dependencies. This small bump resolved it.

### Replaces dropwizard-guicier with dropwizard-guicey
Both libraries handle the integration of the Guice DI framework with dropwizard for you. [dropwizard-guicier](https://github.com/HubSpot/dropwizard-guicier) however hasn't been updated in over two years and doesn't work with DW2. [dropwizard-guicey](https://github.com/xvik/dropwizard-guicey) is much more actively maintained and gives us some neat features that simplify the applications (like auto discovery of jersey extensions).

### Updates the tests from Junit 4 to Junit 5
dropwizard-guicey provides us with a Junit extension for starting our apps for integration tests. We can extend it to configure our apps and replace the Junit 4 app rules we've been using.
This also bumps all the unit tests to use the Junit 5 annotations and error handling constructs. This means we can run all the tests using the [junit platform](https://junit.org/junit5/docs/current/user-guide/#overview-what-is-junit-5)

### CODE REVIEW

**Comment by instantiator** [comment]
Confirmed, this builds successfully with updated dependencies in the local Maven repo. (I've written a short guide to testing this if anyone wants to reproduce.)

File: build.gradle [hunk 1]
```diff
@@ -95,35 +85,30 @@ subprojects {
...
-                'com.hubspot.dropwizard:dropwizard-guicier:1.0.0.6',
+                'ru.vyarus:dropwizard-guicey:5.3.0',
                 'org.reflections:reflections:0.9.10',
-                'org.dhatim:dropwizard-sentry:1.3.9-2'
+                'org.dhatim:dropwizard-sentry:2.0.20'
 
         config 'commons-io:commons-io:2.1'
 
-        verify_event_emitter "uk.gov.ida:verify-event-emitter:2.0.0-82"
+        verify_event_emitter "uk.gov.ida:verify-event-emitter:2.0.0-SNAPSHOT"
```
Comment (instantiator) [inline build.gradle:106]: I've been a bit inconsistent here - but you've acknowledged elsewhere that `SNAPSHOT` references will change to real versions anyway when the time comes.

**Review by idrop — DISMISSED** [review]
I've managed a complete journey with this branch using ida-hub-acceptance-tests, with snapshot dependencies installed in `mavenLocal()`.

We should send this to staging and ZDD!

**Review by JohnWattsGDS — DISMISSED** [review]
Whew!

**Review by JohnWattsGDS — APPROVED** [review]

**Pull request merged**